mozilla_django_oidc/middleware.py

```diff
@@ -0,0 +1,82 @@
+try:
+    from urllib import urlencode
+except ImportError:
+    from urllib.parse import urlencode
+
+import time
+
+import django
+from django.core.urlresolvers import reverse
+from django.http import HttpResponseRedirect
+from django.utils.crypto import get_random_string
+
+from mozilla_django_oidc.utils import absolutify, import_from_settings
+
+
+# Django 1.11 makes changes to how middleware work. In Django 1.11+, we want to
+# use the mixin so that our middleware works as is.
+if tuple(django.VERSION[0:2]) >= (1, 10):
+    from django.utils.deprecation import MiddlewareMixin
+else:
+    class MiddlewareMixin(object):
+        pass
+
+
+# FIXME(willkg): This doesn't appear to be used anywhere.
+def logout_url():
+    """Log out the user from Auth0."""
+    url = 'https://' + import_from_settings('OIDC_OP_DOMAIN') + '/v2/logout'
+    url += '?' + urlencode({
+        'returnTo': import_from_settings('LOGOUT_REDIRECT_URL', '/'),
+        'client_id': import_from_settings('OIDC_RP_CLIENT_ID')
+    })
+    return url
+
+
+class RefreshIDToken(MiddlewareMixin):
+    """Renews id_tokens after expiry seconds
+
+    For users authenticated with an id_token, we need to check that it's still
+    valid after a specific amount of time and if not, force them to
+    re-authenticate silently.
+
+    """
+    def process_request(self, request):
+        if ((request.method == 'GET' and
```

Comment:
Because the middleware now redirects to the authorization endpoint if the token has expired, pretty sure we need to restrict it to just GET requests.

Does that sound right? Are there other methods that will work with the redirects?